SalesforceSDK/Analytics/Model/InstrumentationEvent.cs

```diff
@@ -0,0 +1,237 @@
+        [JsonProperty]
+        public int SequenceId => _sequenceId;
+        [JsonProperty]
+        public string SenderId => _senderId;
+        [JsonProperty]
+        public JObject SenderContext => _senderContext;
+        [JsonProperty]
+        public SchemaType Schema => _schemaType;
+        [JsonProperty]
+        public EventType Event => _eventType;
+        [JsonProperty]
+        public ErrorType Error => _errorType;
+        [JsonProperty]
+        public string ConnectionType => _connectionType;
+        [JsonProperty]
+        public DeviceAppAttributes DeviceAppAttributes => _deviceAppAttributes;
+        [JsonProperty]
+        public string SenderParentId => _senderParentId;
+        [JsonProperty]
+        public long SessionStartTime => _sessionStartTime;
+        [JsonProperty]
+        public JObject Page => _page;
+        [JsonProperty]
+        public JObject PreviousPage => _previousPage;
+        [JsonProperty]
+        public JObject Marks => _marks;
+
+        public InstrumentationEvent(JObject json)
+        {
+            json?.ToObject<InstrumentationEvent>();
+        }
+
+        public JObject ToJson()
+        {
+            var json = new JObject
+            {
+                { EVENT_ID_KEY, _eventId },
+                { START_TIME_KEY, _startTime },
+                { END_TIME_KEY, _endTime },
+                { NAME_KEY, _name },
+                { ATTRIBUTES_KEY, _attributes},
+                { SESSION_ID_KEY, _sessionId},
+                { SEQUENCE_ID_KEY, _sessionId},
+                { SENDER_ID_KEY, _senderId},
+                { SENDER_CONTEXT_KEY, _senderContext},
+                { SCHEMA_TYPE_KEY, _schemaType.ToString()},
+                { EVENT_TYPE_KEY, _eventType.ToString()},
+                { ERROR_TYPE_KEY, _errorType.ToString()},
+                { CONNECTION_TYPE_KEY, _connectionType},
+                { DEVICE_APP_ATTRIBUTES_KEY, _deviceAppAttributes.ToJson()},
+                { SENDER_PARENT_ID_KEY, _senderParentId},
+                { SESSION_START_TIME_KEY, _sessionStartTime },
+                { PAGE_KEY, _page},
+                { PREVIOUS_PAGE_KEY, _previousPage},
+                { MARKS_KEY, _marks}
+            };
+
+            return json;
+        }
+
+        public override bool Equals(Object obj)
+        {
+            if (!(obj is InstrumentationEvent))
+            {
+                return false;
+            }
+            var instrumentationEvent = (InstrumentationEvent) obj;
+            if (string.IsNullOrEmpty(EventId))
+            {
+                return false;
+            }
+            if (EventId.Equals(instrumentationEvent.EventId))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public override int GetHashCode()
+        {
+            return EventId.GetHashCode();
+        }
+
+        /*
+        * Represents the type of event being logged.
```

Comment:
Instead of commenting with `/*/` use `///` that way consumers can see your method and property comments